tools/generate_tables_sql_and_testNames.php

```diff
@@ -23,96 +30,210 @@
+                        ."UPDATE CURRENT_TIMESTAMP,\n"
+                        ."`Data_entry_completion_status` "
+                        ."enum('Incomplete','Complete') "
+                        ."NOT NULL default 'Incomplete',\n";
+            $sql_query_statement = "CREATE TABLE `$sql_query_table_name` (\n";
+            // created first five query entries.
+            array_push(
+                $sql_query_columns,
+                "`CommentID` varchar(255) NOT NULL default '',\n"
+            );
+            array_push(
+                $sql_query_columns,
+                "`UserID` varchar(255) default NULL,\n"
+            );
+            array_push(
+                $sql_query_columns,
+                "`Examiner` varchar(255) default NULL,\n"
+            );
+            array_push(
+                $sql_query_columns,
+                "`Testdate` timestamp DEFAULT CURRENT_TIMESTAMP 
+                ON UPDATE CURRENT_TIMESTAMP,\n"
+            );
+            array_push(
+                $sql_query_columns,
+                "`Data_entry_completion_status` enum('Incomplete',
+                'Complete') NOT NULL default 'Incomplete',\n"
+            );
             break;
-            case "page":
-                $pages[] = $bits[2];
-                continue;
-
-            //no SQL need be generated.
-            case "title":
-                $title = $bits[1];
-            case "header":
-                continue;
+        case "page":
+            $pages[] = $bits[2];
+            continue;
+        //no SQL need be generated.
+        case "title":
+            $title = $bits[1];
+        case "header":
+            continue;
             break;
-
-            //generate specific column definitions for specific types of HTML elements
-            default:
-                if($bits[1] == "") {
-                    continue;
-                }
-                if($bits[0]=="select"){
-                    $bits[0]=enumizeOptions($bits[3], isset($tablename) ? $tablename : null, $bits[1]);
-                } else if($bits[0]=="selectmultiple"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="textarea"){
-                    $bits[0]="text";
-                } else if($bits[0]=="text"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="checkbox") {
-                    $bits[0]="varchar(255)";
-                } else if ($bits[0]=="static") {
-                    $bits[0]="varchar(255)";
-                }
-
-                $bits[2]=htmlspecialchars($bits[2]);
-                $output.="`$bits[1]` $bits[0] default NULL,\n";
+        //generate specific column definitions for specific types of HTML elements
+        default:
+            if ($bits[1] == "") {
+                continue;
+            }
+            if ($bits[0]=="select") {
+                $bits[0] = enumizeOptions(
+                    $bits[3],
+                    isset($tablename)
```

Comment:
`??` idiom would be better here

Reply:
Awesome fixed.

Reply:
I believe it should read `$tablename ?? null`. No `isset`